### ISSUE TITLE
Video Cache Not Clearing as Expected or Unable to Verify Clearing - Android

### ISSUE THREAD

**Issue by oshriza** [FeatherJiang/react-native-video-cache-control#5]
Hi :)

how can I check if the cached video is really cleared?
I'm trying to clear the video cache using `clearAllCache` but when I subsequently check the cache status with `isCached`, it's still returning `true`.

**Expected behavior:**
I expect `isCached` to return `false` after calling `clearAllCache`.

**Environment:**
react-native": "0.70.8",
"react-native-video-cache-control": "^1.2.1"

Is there something I'm missing, or is this something that has not been implemented yet? 
If so, do you have any idea how to check this?

Thanks a lot! :)

**Comment by FeatherJiang** [comment]
please give me a can reproduction example or code, i will check it.
and please tell me ios or android?

**Comment by oshriza** [comment]
Is about Android (title updated).

ok, I wanted to send you the code, and then I thought about checking one more time.
Here is what I found, It appears that the behavior of the caching is slightly inconsistent. Here's what I observed:

1. **First Conversion (Not Cached Yet):**
    The URL initially gets converted to a localhost URL (http://127.0.0.1), likely indicating the cache is not yet ready.

2. **Second Conversion (Cached):**
    After the cache is ready, the conversion URL changes to a path within the app's "private" cache directory (`file:///data/user/0/com.myclientapp/cache/video-cache/...`), and `isCached` was `true`.

3. **Clearing Cache Issue:**
    When calling `clearAllCache`  with the "private" cache URL, `isCached` still returns `true`. This seems to be inconsistent behavior.

4. **Different Cache Path (More "Public") - **WORKS** as expected:**
    Later, I noticed the conversion URL changing to a more "public" cache directory (`file:///storage/emulated/0/Android/data/com.myclientapp/cache/video-cache/...`). With this URL, the cache folder is visible on the device, and `clearAllCache`  behaves as  **expected**,  and setting `isCached`  to `** false.**`

I'm unsure why these inconsistencies are occurring, particularly the difference between the two cache paths and the behavior of `clearAllCache`. It might be related to the state of the device's storage or the way the caching library is configured.

Do you have any idea about that?
**<h3>Thank you for all your work!!!🙏</h3>**

**Here is my component code:**
```
import React, { FC, useEffect, useRef, useState } from 'react';
import { ActivityIndicator, Dimensions, Text, ViewStyle } from 'react-native';
import Video from 'react-native-video';
import { clearCache, convertAsync, isCached } from 'react-native-video-cache-control';

import { AppEvent } from './types';

const $backgroundVideo: ViewStyle = {
  position: 'absolute',
  top: 0,
  left: 0,
  bottom: 0,
  right: 0,
};
const EventVideo: FC<{ event: AppEvent; activeIndex: number; index: number }> = React.memo(
  ({ event, activeIndex, index }) => {
    const [proxyURL, setProxyURL] = useState();
    const playerRef = useRef(null);
    const videoEndedRef = useRef(false);
    const [loading, setLoading] = useState(true);
    const [isVideoCached, setIsVideoCached] = useState(false);

    useEffect(() => {
      const getProxyURL = async () => {
        if (proxyURL) {
          console.log('Proxy URL already exists, skipping conversion...');
          return; // Return early if proxyURL already exists
        }
        try {
          console.log('Converting URL...');
          const url = await convertAsync({ url: event.videoUrl });
          console.log('==== Original URL: ====', event.videoUrl);
          console.log('==== Proxy URL: ====', url);
          setProxyURL(url);
          setLoading(false);
        } catch (error) {
          console.error('Error in conversion:', error);
          setLoading(false);
        }
      };

      getProxyURL();
    }, [event.videoUrl, proxyURL]); // Added proxyURL to the dependency array

    function checkReady() {
      console.log('ready to play');
      console.log('event id: ' + event.id);
    }

    function handleVideoEnd() {
      videoEndedRef.current = true;
      console.log('Video ended');
      // Clear the cache if the video has ended twice
      clearAllCache(() => {
        // Add a delay to ensure the cache is cleared, then check if the video is cached
        setTimeout(() => {
          checkIfCached();
        }, 20000); // Adjust the delay as needed
      });
    }
    const checkIfCached = async () => {
      try {
        const cached = await isCached({ url: event.videoUrl });
        console.log('Is video cached:', cached); // Should log false if cache is cleared
        setIsVideoCached(cached);
      } catch (error) {
        console.log(error);
      }
    };
    const clearAllCache = async (callback) => {
      try {
        console.log('Clearing cache...');
        await clearCache();
        console.log('Cache cleared');
        if (callback) {
          callback(); // Call the callback if it's provided
        }
      } catch (error) {
        console.log(error);
      }
    };

    useEffect(() => {
      if (videoEndedRef.current) {
        console.log('Component re-rendered after video end');
      }
    });

    return loading ? (
      <ActivityIndicator size="large" color="#0000ff" />
    ) : (
      <Video
        source={{ uri: proxyURL }}
        style={$backgroundVideo}
        poster={event.posterUrl}
        resizeMode="cover"
        posterResizeMode="cover"
        playInBackground={false}
        playWhenInactive={false}
        paused={index !== activeIndex}
        onError={(error) => {
          console.error('Video loading error', error);
        }}
        onReadyForDisplay={checkReady}
        ref={playerRef} // Set the ref
        onEnd={handleVideoEnd} // Handle end of video
      />
    );
  }
);

export default EventVideo;
```

**Comment by FeatherJiang** [comment]
I fix this bug and please update to 1.2.2
